mlir/include/mlir/Dialect/DLTI/DLTI.h

```diff
@@ -124,6 +126,141 @@ class DataLayoutSpecAttr
   static constexpr StringLiteral name = "builtin.data_layout_spec";
 };
 
+//===----------------------------------------------------------------------===//
+// TargetSystemDescSpecAttr
+//===----------------------------------------------------------------------===//
+
+/// A system description attribute is a list of device descriptors, each
+/// having a uniq device ID
+class TargetSystemDescSpecAttr
```

Comment:
Why can't this attribute be defined with TableGen?

Reply:
I was following the same mechanism as it is currently in DLTI.